guide/src/class.md

```diff
@@ -140,19 +143,19 @@ Rules for the `new` method:
 * If no method marked with `#[new]` is declared, object instances can only be created
   from Rust, but not from Python.
 * The first parameter is the raw object and the custom `new` method must initialize the object
-  with an instance of the struct using `init` method. The type of the object may be the type object of
+  with an instance of the struct using the `init` method. The type of the object may be the type object of
   a derived class declared in Python.
-* The first parameter implicitly has type `&PyRawObject`.
-* For details on `parameter-list`, see the documentation of `Method arguments` section.
-* The return type must be `PyResult<T>` for some `T` that implements `IntoPyObject`. Usually, `T` will be `MyType`.
+* The first parameter must have type `&PyRawObject`.
+* For details on the parameter list, see the `Method arguments` section below.
+* The return value must be `T` or `PyResult<T>` where `T` is ignored, so it can
```

Comment:
This changes the actual meaning of the sentence. What does it mean that `T` is ignored?

Are you saying that nothing uses the return value of this function unless it's a `PyResult<T>`, which is used to throw exceptions, but not to return a value?

Reply:
> What does it mean that T is ignored?

Agreed

Reply:
> Are you saying that nothing uses the return value of this function unless it's a PyResult<T>, which is used to throw exceptions, but not to return a value?

Yes, exactly. It's the same as Python's `__init__` in that regard.

See https://github.com/PyO3/pyo3/blob/master/pyo3-derive-backend/src/pymethod.rs#L192 where the result is matched as `Ok(_)`.

The old text probably refers to an earlier version's behavior.